### PR TITLE
add new comment feature on article page

### PR DESCRIPTION
**Pull request by AudeNgD**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import ArticlePage from "./components/ArticlePage";
 import "./App.css";
 
 function App() {
-  const [currentUser, SetCurrentUser] = useState("");
+  const [currentUser, SetCurrentUser] = useState("jessjelly");
```
Comment (nc-christian-dutton) [inline src/App.jsx:11]: setCurrentUser should be camelCase syntax - good idea to hard code the user!

File: src/components/ArticlePage.jsx [hunk 2]
```diff
@@ -60,20 +62,48 @@ export default function ArticlePage() {
...
+  }
+
+  function addNewCommentToList(comment) {
+    fetchComments(articleId).then(() => {
+      comment.votes = 0;
+      setComments((currentComments) => {
+        return [comment, ...currentComments];
+      });
+    });
+  }
```
Comment (nc-christian-dutton) [inline src/components/ArticlePage.jsx:76]: You shouldn't need to do fetchComments again here, if your useEffect is already fetching the comments and setting them as state then you have access to currentComments already through setComments, then a change in the state should cause a re-render

File: src/components/NewCommentForm.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  function addNewComment(event) {
+    event.preventDefault();
+    isDisabled(true);
+    //setTimeout(isDisabled(false), 5000);
+    const newComment = { author: currentUser, body: userComment };
+    addNewCommentToList(newComment);
+    postNewComment(newComment, article_id);
+    setUserComment("");
```
Comment (nc-christian-dutton) [inline src/components/NewCommentForm.jsx:22]: Optimistically done, nice!

File: src/components/NewCommentForm.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  return disabled ? (
+    <p>
+      Thanks for leaving a comment! People like you make our community grow!
+    </p>
+  ) : (
```
Comment (nc-christian-dutton) [inline src/components/NewCommentForm.jsx:29]: very nice feature ☝️

File: src/App.jsx (hunk 1 above)
Comment (AudeNgD) [reply src/App.jsx:11]: OK- updated

File: src/components/ArticlePage.jsx (hunk 2 above)
Comment (AudeNgD) [reply src/components/ArticlePage.jsx:76]: OK - refactored

**Pull request merged**